docs/_posts/2018-01-15-jekyll-3-7-1-released.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "Jekyll 3.7.1 Released"
+date: 2018-01-15 11:21:00 +0530
+author: ashmaroli
+version: 3.7.1
+categories: [release]
+---
+
+Close on the heels of shipping 3.7.0, we were informed of a couple of
+regressions due to the changes made in that release. In due time, Team Jekyll
+set out to address those issues as early as possible.
+
+Days later, here we're announcing 3.7.1 that aims to set things in order. The
+highlights being:
+
+  * A major regression in 3.7.0 was that when a Front Matter Default was
+    configured with a `scope["path"]` set to a directory, Jekyll would scan
+    that directory for any subfolders and files, for each document in that
+    `path`.
+    Though this is intended, it increases build times in proportion to the size
+    of the said directory. <br />
+    We addressed this by having Jekyll scan the directory path only if the user
+    explicitly configures the `scope["path"]` using wildcards. <br />
+    Read our documentation for more details. <br />
+    A huge shout-out to @mmistakes for bringing this to our notice and
+    additionally providing with a test repository to aid in resolving the issue.
+  * Another regression reported was related to our "Custom collections
+    directory" feature introduced in 3.7.0. <br />
+    Users setting `collection_dir` to a certain directory would have *altered*
+    paths to their posts still at the root of their site's source. This
+    roughly translated to 404 errors on URLs to their posts. <br />
+    Props to @localheinz for asserting this regression to our notice. <br />
+    We decided to resolve this by having Jekyll ignore posts and drafts at the
+    root of the site's source directory *if the user customizes the
+    `collection_dir` setting.* <br />
```

Comment:
Moving `_posts` and `_drafts`out of the source directory is a tempting fix but can't it have consequences on how some plugins work ?

Reply:
I'd love it if you can point me to at least one popular plugin that could break.. (not a challenge.., I'd really like to see if there's a workaround / resolution..)

Reply:
I was thinking of paginationv2, jekyll-admin, jekyll-feed to name a few

Reply:
IIRC, `jekyll-feed`, `jekyll-sitemap` calls to posts via `{% for post in site.posts %}`. Since `Document` attributes do not change at all, (`relative_path` is the same irrespective of the `collection_dir`) setting,  I doubt these plugins are going to break..
But what is evident from jekyll/jekyll-feed#202 is that **plugins will break if the posts are at the root directory when a custom collection_dir has been set.**